### PR TITLE
ONL-4193 Dropdown loading state

### PR DESCRIPTION
**Pull request by jaimeperezpedra**
Add the loading state to the dropdown and dropdown-search.

I discussed with @matallo that is mandatory if is loading we cannot display the 'no results found' field

### CODE REVIEW

File: src/components/ec-dropdown-search/ec-dropdown-search.vue [hunk 1]
```diff
@@ -62,7 +62,15 @@
               :title="noResultsText"
             >{{ noResultsText }}</li>
           </slot>
+          <ec-loading
```
Comment (mcibique) [inline src/components/ec-dropdown-search/ec-dropdown-search.vue]: wrap this in `<li>`

File: src/components/ec-dropdown-search/ec-dropdown-search.vue [hunk 2]
```diff
@@ -348,6 +361,10 @@ $ec-dropdown-search-item-delimiter-size: 1px !default;
     @include ec-dropdown-search-item;
   }
 
+  &__loading {
+    height: 40px;
```
Comment (mcibique) [inline src/components/ec-dropdown-search/ec-dropdown-search.vue]: `height: $ec-dropdown-search-item-height`

File: src/components/ec-dropdown/ec-dropdown.story.js [hunk 3]
```diff
@@ -50,6 +50,9 @@ stories.add('basic', () => ({
     disabled: {
       default: boolean('disabled', false),
     },
+    isLoading: {
+      default: boolean('isLoading', false),
+    },
```
Comment (mcibique) [inline src/components/ec-dropdown/ec-dropdown.story.js]: could you add this to the ec-dropdown-search.story.js too?

File: src/components/ec-dropdown-search/ec-dropdown-search.spec.js [hunk 4]
```diff
@@ -136,6 +136,11 @@ describe('EcDropdownSearch', () => {
     expect(wrapper.find('.ec-dropdown-search__search-area').element).toMatchSnapshot();
   });
 
+  it('should not show the items when the loading is set to true', () => {
+    const wrapper = mountDropdownSearch({ isLoading: true });
```
Comment (mcibique) [inline src/components/ec-dropdown-search/ec-dropdown-search.spec.js]: you have to pass some items into the dropdown to verify that:
```
const wrapper = mountDropdownSearch({ items, isLoading: true });
```

File: src/components/ec-dropdown/ec-dropdown.spec.js [hunk 5]
```diff
@@ -142,6 +142,11 @@ describe('EcDropdown', () => {
         const wrapper = mountDropdownSingleValue({ disabled: true });
         expect(wrapper.element).toMatchSnapshot();
       });
+
+      it('should not show the items when the loading is set to true', () => {
+        const wrapper = mountDropdownSingleValue({ isLoading: true });
```
Comment (mcibique) [inline src/components/ec-dropdown/ec-dropdown.spec.js]: you have to pass some items into the dropdown to verify that:
```
const wrapper = mountDropdownSingleValue({ items, isLoading: true });
```

File: src/components/ec-dropdown/ec-dropdown.spec.js [hunk 6]
```diff
@@ -142,6 +142,11 @@ describe('EcDropdown', () => {
         const wrapper = mountDropdownSingleValue({ disabled: true });
         expect(wrapper.element).toMatchSnapshot();
       });
+
+      it('should not show the items when the loading is set to true', () => {
```
Comment (mcibique) [inline src/components/ec-dropdown/ec-dropdown.spec.js]: We need to add the same test into `describe('multiple value')`.

File: src/components/ec-dropdown-search/ec-dropdown-search.vue [hunk 7]
```diff
@@ -62,7 +62,16 @@
               :title="noResultsText"
             >{{ noResultsText }}</li>
           </slot>
+          <li v-if="isLoading">
```
Comment (matallo) [inline src/components/ec-dropdown-search/ec-dropdown-search.vue]: Im'm curious, why not putting this at the top and then using the `v-if-else`?

Comment (jaimeperezpedra) [reply src/components/ec-dropdown-search/ec-dropdown-search.vue]: Yeah, good call!

Comment (jaimeperezpedra) [reply src/components/ec-dropdown-search/ec-dropdown-search.vue]: Done!

Comment (matallo) [reply src/components/ec-dropdown-search/ec-dropdown-search.vue]: Well I meant: 

- if is loading
- else if is empty (it is implied it is not loading)
...

in that order, thus you don't have to do the is not loading for the empty state by reordering the conditions

Comment (stavros-tomas) [reply src/components/ec-dropdown-search/ec-dropdown-search.vue]: @matallo let me know if this ok for you

**Pull request merged**